app/screens/map/mapSelector.js

```diff
@@ -1,11 +1,23 @@
+import pickBy from 'lodash/pickBy';
+
 import { createSelector, createStructuredSelector } from 'reselect';
 
 function unitsSelector(state) {
   return state.data.units;
 }
 
-const markersSelector = createSelector(
+function cityFilterSelector(state) {
+  return state.filters.city;
+}
+
+const filteredUnitsSelector = createSelector(
   unitsSelector,
+  cityFilterSelector,
+  (units, cityFilter) => pickBy(units, unit => cityFilter === '' || unit.city === cityFilter)
```

Comment:
- [ ] A slight performance improvement could be:
```javascript
(units, cityFilter) => cityFilter === '' ? units : pickBy(units, unit => unit.city === cityFilter)
```
But don't know if that has much of an affect.